doc/release/1.8.0-notes.rst

```diff
@@ -50,80 +86,808 @@ New features
+to ``zpk2sos``.
+
+``savgol_coeffs`` and ``savgol_filter`` now work for even window lengths.
+
+Added the Chirp Z-transform and Zoom FFT available as `scipy.signal.CZT` and
+`scipy.signal.ZoomFFT`.
 
 `scipy.sparse` improvements
 ===========================
 
+An array API has been added to facilitate the ecosystem-wide transition away
+from the matrix API. We recommend that new work uses these array objects:
+``bsr_array``, ``coo_array``, etc. Please refer to the `scipy.sparse`
+docstring for more information.
+
+``maximum_flow`` introduces optional keyword only argument, ``method``
+which accepts either, ``'edmonds-karp'`` (Edmonds Karp algorithm) or
+``'dinic'`` (Dinic's algorithm). Moreover, ``'dinic'`` is used as default
+value for ``method`` which means that Dinic's algorithm is used for computing
+maximum flow unless specified. See, the comparison between the supported
+algorithms in
+`this comment <https://github.com/scipy/scipy/pull/14358#issue-684212523>`_.
+
+Parameters ``atol``, ``btol`` now default to 1e-6 in
+`scipy.sparse.linalg.isolve.lsmr` to match with default values in
+`scipy.sparse.linalg.isolve.lsqr`.
+
+Add the Transpose-Free Quasi-Minimal Residual algorithm (TFQMR) for general
+nonsingular non-Hermitian linear systems in `scipy.sparse.linalg.isolve.tfqmr`.
+
+The sparse SVD library PROPACK is now vendored with SciPy, and an interface is
+exposed via `scipy.sparse.svds` with ``solver='PROPACK'``. For some problems,
+this may be faster and/or more accurate than the default, ARPACK.
+
+``sparse.linalg`` now has a nonzero initial guess option, which may be
+specified as ``x0 = 'Mb'``.
+
+Some performance and behavior improvements for
+`scipy.sparse.csgraph.maximum_flow`.
+
+The ``trace`` method has been added for sparse matrices.
 
 
 `scipy.spatial` improvements
 ============================
 
+`scipy.spatial.transform.Rotation` now supports item assignment and has a new
+``concatenate`` method. `scipy.spatial.distance.minkowski` now also supports
+``0<p<1``.
+
+Add `scipy.spatial.distance.kulczynski1` in favour of
+`scipy.spatial.distance.kulsinski` which will be deprecated in the next
+release.
+
+We now allow `p>0` for the Minkowski distance.
 
 `scipy.special` improvements
 ============================
 
+The new function `scipy.special.log_expit` computes the logarithm of the
+logistic sigmoid function. The function is formulated to provide accurate
+results for large positive and negative inputs, so it avoids the problems
+that would occur in the naive implementation ``log(expit(x))``.
+
+A suite of five new functions for elliptic integrals: 
+``scipy.special.ellipr{c,d,f,g,j}``. These are the
+`Carlson symmetric elliptic integrals <https://dlmf.nist.gov/19.16>`_, which
+have computational advantages over the classical Legendre integrals. Previous
+versions included some elliptic integrals from the Cephes library
+(``scipy.special.ellip{k,km1,kinc,e,einc}``) but was missing the integral of
```

Comment:
I think this wasn't discussed in the PR reviews: @duetosymmetry it sounds like there's duplicate functionality now, and the new functions are better and have complex support. Is that really the case and if so, should we at least at some `.. note::`s (or perhaps even a deprecation) in the Cephes based functions to prefer the new functions?

Reply:
@rgommers that's right, the new functions should be a strict superset of what was previously possible. Is there a policy written somewhere about how scipy dev handled this in the past?

I added the relationships between the old and new functions into the `Notes` sections of the old docs.

I added tests for equivalence between the old and new functions, but only for three out of the five old functions (k, km1, and e). I would like to add tests for the other two before replacing their implementations with the new functions.

Related issue that I opened for the future: #15151

Reply:
Thanks, that's helpful. There's a policy at http://scipy.github.io/devdocs/dev/core-dev/index.html#deprecations. In this case, if it's a strict superset and the signatures are compatible, we probably should have just reused the old function names and replaced the implementations. Or if they're different, we can deprecate the old names in the next release.

Reply:
Thanks for the policy link. I was editing my comment, possibly at the same time as you were responding. I think convenience functions for the classical function definitions are desirable for 95% of people who will use elliptic integrals. So, this would not deprecate the old API, but just rewrite the implementations of those functions so they use the more general newer functions.

Reply:
That sounds good. Being able to remove a lot of old low-level code would be a win in itself.

Reply:
I actually don't see a concise request for release notes change in this discussion, so decent chance I'll leave this alone for now. If it really needs to change later we can do it before or between release candidates.